nuxeo-runtime/nuxeo-runtime-deploy/src/main/java/org/nuxeo/runtime/deployment/preprocessor/install/commands/AppendCommand.java

```diff
@@ -94,4 +97,23 @@ public String toString(CommandContext ctx) {
         return "append " + ctx.expandVars(src.toString()) + " > " + ctx.expandVars(dst.toString());
     }
 
+    private void append(File srcFile, File dstFile, boolean appendNewLine) throws IOException {
+        String srcExt = FileUtils.getFileExtension(srcFile.getName());
+        String dstExt = FileUtils.getFileExtension(dstFile.getName());
+        if (!StringUtils.equalsIgnoreCase(srcExt, dstExt)) {
+            throw new IOException("Source and destination files have different extensions.");
+        }
+
+        switch (srcExt != null ? srcExt.toLowerCase() : "") {
+        case "json":
```

Comment:
Do you expect this `switch` to grow new cases? Otherwise a simple `if ("json".equalsIgnoreCase(srcExt))` is simpler.